File: reference/metarealm-growth-os-main/backend/app/schemas/content.py
from typing import Literal

from app.schemas.base import CamelModel

ContentPlatform = Literal["linkedin", "x", "instagram"]
ContentStatus = Literal["awaiting_approval", "draft", "scheduled", "published"]
NewsRegion = Literal["morocco", "mena", "web3", "gaming"]


class ContentItemBase(CamelModel):
    title: str
    platform: ContentPlatform
    status: ContentStatus
    scheduled_for: str | None = None
    body: str | None = None
    author: str | None = None


class ContentItemCreate(ContentItemBase):
    pass


class ContentItemUpdate(CamelModel):
    title: str | None = None
    platform: ContentPlatform | None = None
    status: ContentStatus | None = None
    scheduled_for: str | None = None
    body: str | None = None
    author: str | None = None


class ContentItemOut(ContentItemBase):
    topic: str = "general"
    id: str


class NewsItemOut(CamelModel):
    id: str
    title: str
    source: str
    region: NewsRegion
    published_ago: str
    url: str | None = None
    saved: bool = False
    score: int = 0

File: reference/metarealm-growth-os-main/frontend/types/content.ts
/** Content studio + market intelligence domain. */

export type ContentPlatform = "linkedin" | "x" | "instagram";

export type ContentStatus =
  | "awaiting_approval"
  | "draft"
  | "scheduled"
  | "published";

export interface ContentItem {
  id: string;
  saved?: boolean;
  topic?: string;
  title: string;
  platform: ContentPlatform;
  status: ContentStatus;
  scheduledFor?: string;
  /** Draft text — full post or opening excerpt during the mock phase. */
  body?: string;
  author?: string;
}

export type NewsRegion = "morocco" | "mena" | "web3" | "gaming";

export interface NewsItem {
  id: string;
  title: string;
  source: string;
  region: NewsRegion;
  publishedAgo: string;
  url?: string | null;
  saved?: boolean;
  topic?: string;
  score?: number;
}
